src/spdx_tools/spdx/document_utils.py

```diff
@@ -49,9 +49,12 @@ def create_document_without_duplicates(document: Document) -> Document:
 
 
 def create_list_without_duplicates(list_with_potential_duplicates: List[Any]) -> List[Any]:
+    seen_elements = set()
     list_without_duplicates = []
+
     for element in list_with_potential_duplicates:
-        if element not in list_without_duplicates:
+        if element not in seen_elements:
+            seen_elements.add(element)
```

Comment:
Instead of storing the element, you could store the element ID or similar to https://github.com/spdx/tools-python/pull/792 store the `astuple(element)`. This could solve the current issues in the CI